### PR TITLE
Clean up CCA logging

### PR DESCRIPTION
**Pull request by hankewyczz**
~I based this PR off of the `refactor-js-to-ts` branch, so don't merge until that one's done.~ Ready for review

https://trello.com/c/JK7r05uA/278-refactor-backend-to-use-macrosverbose

- Added `http()` log level to `Macros`
- Added `winston` support for log files. 
  - Logs are now saved in the `logs/` directory. There are three files - one for `verbose` level logs and above, one for `log` level logs and above, and one for `warn` level logs and above
- Added `winston-daily-rotate-file` support for log rotation. 

Default logging behavior:
The log levels are:
1. `critical`
2. `error`
3. `warn`
4. `info` (aka `log`)
5. `http`
6. `verbose`

By default, only logs of severity `info` and above are shown on the console output (however, all level logs are saved to the log files). This can be changed by setting the `LOG_LEVEL`.

The console will display all logs of the level set by `LOG_LEVEL` and above (eg. `LOG_LEVEL=warn` will display `warn`, `error`, and `critical` messages).

### CODE REVIEW

File: scrapers/classes/parsersxe/termParser.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import {
   SectionSR,
 } from "../../../types/scraperTypes";
 import { CourseRef, Section } from "../../../types/types";
+import "colors";
```
Comment (andrewydai) [inline scrapers/classes/parsersxe/termParser.ts:23]: do we need to import colors if not using it here

File: scrapers/classes/parsersxe/tests/prereqParser.test.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ jest.mock("../subjectAbbreviationParser");
 const subjectAbbreviationTable = getSubjectAbbreviations();
 
 describe("prereqParser", () => {
-  it("should parse corequisites", () => {
+  it("should parse corequisites", async () => {
```
Comment (andrewydai) [inline scrapers/classes/parsersxe/tests/prereqParser.test.js:9]: Does this need to be changed/was it supposed to part of this pr

**Review by andrewydai — APPROVED** [review]
looks awesome, pretty minor comments :)

File: scrapers/classes/parsersxe/termParser.ts (hunk 1 above)
Comment (hankewyczz) [reply scrapers/classes/parsersxe/termParser.ts]: Nope

File: scrapers/classes/parsersxe/tests/prereqParser.test.js (hunk 2 above)
Comment (hankewyczz) [reply scrapers/classes/parsersxe/tests/prereqParser.test.js]: No/no, it was a remnant of the Typescript PR (removed in the original PR, but not here)

**Pull request merged**